src/condor_kbdd/XInterface.unix.cpp

```diff
@@ -116,7 +117,61 @@ XInterface::TryUser(const char *user)
 	dprintf( D_FULLDEBUG, "Using %s's .Xauthority: \n", passwd_entry->pw_name );
 	return true;
 }
-	
+
+// Newer display managers have moved the location of the .Xauthority
+// file from $HOME into $XDG_RUNTIME_DIR/.xauth_<some_nonce>
+bool
+XInterface::TryUserXDG(const char *user)
+{
+	static char env[1024];
+	static bool need_uninit = false;
+	passwd *passwd_entry;
+	std::string xauth_filename_str; 
+
+	passwd_entry = getpwnam(user);
+	if(passwd_entry == NULL) {
+		// We couldn't find the current user in the passwd file?
+		dprintf( D_FULLDEBUG, 
+				"Current user cannot be found in passwd file.\n" );
+		return false;
+	} else {
+		std::string xdg_runtime_dir_str;
+
+		// find the first file named .xauth_<something> in XDG_RUNTIME_DIR
+		formatstr(xdg_runtime_dir_str, "/var/run/user/%d/", passwd_entry->pw_uid);
+		std::filesystem::path xdg_runtime_dir {xdg_runtime_dir_str};
+		for (const auto &entry: std::filesystem::directory_iterator(xdg_runtime_dir)) {
+			if (entry.path().filename().string().starts_with("xauth_")) {
```

Comment:
Does the `starts_with()` comparison not need `.xauth_` and not `xuath_` or is the `std::filesystem` entry consuming the period or something?